File: 2020/days/01.py
from utils import get_input


def p1(inp):
    n = len(inp)
    ht = {}

    for i in range(n):
        ht[inp[i]] = i

    for i in range(n):
        if 2020 - inp[i] in ht and i != ht[2020 - inp[i]]:
            return inp[i] * (2020 - inp[i])

    return -1


def p2(inp):
    n = len(inp)
    ht = {}

    for i in range(n):
        ht[inp[i]] = i

    for i in range(n):
        target = 2020 - inp[i]

        for j in range(1, n):
            if target - inp[j] in ht:
                return inp[i] * inp[j] * (target - inp[j])
    return -1


inp = list(map(int, get_input()))
print(p1(inp))
print(p2(inp))

File: 2015/days/01.ts
#!/usr/bin/env ts-node-script

import get_input from '../utils/getline';

let input: string[] = get_input();

const goToFloor = (input: string, stopAt) => {
    let floor: number = 0;

    for (let i = 0; i < input.length; ++i) {
        if (input[i] === '(') {
            ++floor;
        }
        else if (input[i] === ')') {
            --floor;
        }
        
        if (floor === stopAt) {
            return i + 1;
        }
    }

    return floor;

}
const p1 = (inp: string[]): number => {
    return goToFloor(inp[0], null);
}

const p2 = (inp: string[]): number => {
    return goToFloor(inp[0], -1);
}

console.log(p1(input));
console.log(p2(input));